# 1. Loading Dataset

In [6]:
import os

database_size = int(int(os.path.getsize("database.sqlite"))/1e+6)
print("Size of the database.sqlite file is : ", database_size," MBs")

Size of the database.sqlite file is :  313  MBs


# 2 . Viewing Tables

In [29]:
from sqlite3 import connect
DATABASE_NAME="database.sqlite"


conn = connect(DATABASE_NAME)
cursor = conn.cursor()
cursor.execute("SELECT name FROM sqlite_schema WHERE type='table' AND name not LIKE 'sqlite_%'; ")
tables = cursor.fetchall()
print("All Tables")
print("*"*100)
print(*tables,sep="\n")
print("*"*100)

cursor.close()
conn.close()

All Tables
****************************************************************************************************
('Player_Attributes',)
('Player',)
('Match',)
('League',)
('Country',)
('Team',)
('Team_Attributes',)
****************************************************************************************************


In [30]:
import pandas as pd


def GetTable_DataFrame(table_name: str) -> pd.DataFrame:
    """
    Description:
        This function converts SQL based table into a pandas dataframe

    Args:
        table_name

    Returns:
        dataframe if success, else False

    Raises:
        RaiseException: if table_name does not exist
        RaiseException: if database does not exist


    """

    conn = connect(DATABASE_NAME)
    try:
        sql_statement=f"SELECT * from {table_name};"
        return pd.read_sql(con=conn,sql=sql_statement)
    except Exception as e:
        print(f"Error Failed to fetch Table {table_name} Reasons: {e}")
        return False
    finally:
        conn.close()

SELECT
 Player_Attributes.id,
 Player_Attributes.player_api_id,
 Player_Attributes.player_fifa_api_id,
 player.player_name,
 player.height,
 player.weight,
 AVG(Player_Attributes.overall_rating),
 AVG(Player_Attributes.potential),
 Player_Attributes.preferred_foot,
 Player_Attributes.attacking_work_rate,
 Player_Attributes.defensive_work_rate,
 AVG(Player_Attributes.crossing),
 AVG(Player_Attributes.finishing),
 AVG(Player_Attributes.heading_accuracy),
 AVG(Player_Attributes.short_passing),
 AVG(Player_Attributes.volleys),
 AVG(Player_Attributes.dribbling),
 AVG(Player_Attributes.curve),
 AVG(Player_Attributes.free_kick_accuracy),
 AVG(Player_Attributes.long_passing),
 AVG(Player_Attributes.ball_control),
 AVG(Player_Attributes.acceleration),
 AVG(Player_Attributes.sprint_speed),
 AVG(Player_Attributes.agility),
 AVG(Player_Attributes.reactions),
 AVG(Player_Attributes.balance),
 AVG(Player_Attributes.shot_power),
 AVG(Player_Attributes.jumping),
 AVG(Player_Attributes.stamina),
 AVG(Player_Attributes.strength),
 AVG(Player_Attributes.long_shots),
 AVG(Player_Attributes.aggression),
 AVG(Player_Attributes.interceptions),
 AVG(Player_Attributes.positioning),
 AVG(Player_Attributes.vision),
 AVG(Player_Attributes.penalties),
 AVG(Player_Attributes.marking),
 AVG(Player_Attributes.standing_tackle),
 AVG(Player_Attributes.sliding_tackle),
 AVG(Player_Attributes.gk_diving),
 AVG(Player_Attributes.gk_handling),
 AVG(Player_Attributes.gk_kicking),
 AVG(Player_Attributes.gk_positioning),
 AVG(Player_Attributes.gk_reflexes)
 
 
From 
	Player_Attributes
INNER JOIN
	Player
ON
	Player_Attributes.player_api_id = Player.player_api_id
GROUP BY
	Player_Attributes.player_api_id,
	Player_Attributes.player_fifa_api_id
	
ORDER BY
	 Player_Attributes.id;


**2.1 Player Attribute Table**

In [31]:
GetTable_DataFrame("Player_Attributes")

id  player_fifa_api_id  player_api_id                 date  \
0            1              218353         505942  2016-02-18 00:00:00   
1            2              218353         505942  2015-11-19 00:00:00   
2            3              218353         505942  2015-09-21 00:00:00   
3            4              218353         505942  2015-03-20 00:00:00   
4            5              218353         505942  2007-02-22 00:00:00   
...        ...                 ...            ...                  ...   
183973  183974              102359          39902  2009-08-30 00:00:00   
183974  183975              102359          39902  2009-02-22 00:00:00   
183975  183976              102359          39902  2008-08-30 00:00:00   
183976  183977              102359          39902  2007-08-30 00:00:00   
183977  183978              102359          39902  2007-02-22 00:00:00   

        overall_rating  potential preferred_foot attacking_work_rate  \
0                 67.0       71.0          right              medium   
1                 67.0       71.0          right              medium   
2                 62.0       66.0          right              medium   
3                 61.0       65.0          right              medium   
4                 61.0       65.0          right              medium   
...                ...        ...            ...                 ...   
183973            83.0       85.0          right              medium   
183974            78.0       80.0          right              medium   
183975            77.0       80.0          right              medium   
183976            78.0       81.0          right              medium   
183977            80.0       81.0          right              medium   

       defensive_work_rate  crossing  ...  vision  penalties  marking  \
0                   medium      49.0  ...    54.0       48.0     65.0   
1                   medium      49.0  ...    54.0       48.0     65.0   
2                   medium      49.0  ...    54.0       48.0     65.0   
3                   medium      48.0  ...    53.0       47.0     62.0   
4                   medium      48.0  ...    53.0       47.0     62.0   
...                    ...       ...  ...     ...        ...      ...   
183973                 low      84.0  ...    88.0       83.0     22.0   
183974                 low      74.0  ...    88.0       70.0     32.0   
183975                 low      74.0  ...    88.0       70.0     32.0   
183976                 low      74.0  ...    88.0       53.0     28.0   
183977                 low      74.0  ...    88.0       53.0     38.0   

        standing_tackle  sliding_tackle  gk_diving  gk_handling  gk_kicking  \
0                  69.0            69.0        6.0         11.0        10.0   
1                  69.0            69.0        6.0         11.0        10.0   
2                  66.0            69.0        6.0         11.0        10.0   
3                  63.0            66.0        5.0         10.0         9.0   
4                  63.0            66.0        5.0         10.0         9.0   
...                 ...             ...        ...          ...         ...   
183973             31.0            30.0        9.0         20.0        84.0   
183974             31.0            30.0        9.0         20.0        73.0   
183975             31.0            30.0        9.0         20.0        73.0   
183976             32.0            30.0        9.0         20.0        73.0   
183977             32.0            30.0        9.0          9.0        78.0   

        gk_positioning  gk_reflexes  
0                  8.0          8.0  
1                  8.0          8.0  
2                  8.0          8.0  
3                  7.0          7.0  
4                  7.0          7.0  
...                ...          ...  
183973            20.0         20.0  
183974            20.0         20.0  
183975            20.0         20.0  
183976            20.0         20.0  
183977        

**2.2 Player Table**

In [32]:
GetTable_DataFrame("Player")

id  player_api_id          player_name  player_fifa_api_id  \
0          1         505942   Aaron Appindangoye              218353   
1          2         155782      Aaron Cresswell              189615   
2          3         162549          Aaron Doran              186170   
3          4          30572        Aaron Galindo              140161   
4          5          23780         Aaron Hughes               17725   
...      ...            ...                  ...                 ...   
11055  11071          26357       Zoumana Camara                2488   
11056  11072         111182         Zsolt Laczko              164680   
11057  11073          36491            Zsolt Low              111191   
11058  11074          35506  Zurab Khizanishvili               47058   
11059  11075          39902   Zvjezdan Misimovic              102359   

                  birthday  height  weight  
0      1992-02-29 00:00:00  182.88     187  
1      1989-12-15 00:00:00  170.18     146  
2      1991-05-13 00:00:00  170.18     163  
3      1982-05-08 00:00:00  182.88     198  
4      1979-11-08 00:00:00  182.88     154  
...                    ...     ...     ...  
11055  1979-04-03 00:00:00  182.88     168  
11056  1986-12-18 00:00:00  182.88     176  
11057  1979-04-29 00:00:00  180.34     154  
11058  1981-10-06 00:00:00  185.42     172  
11059  1982-06-05 00:00:00  180.34     176  

[11060 rows x 7 columns]

**2.3 Match Table**

In [33]:
GetTable_DataFrame("Match")

id  country_id  league_id     season  stage                 date  \
0          1           1          1  2008/2009      1  2008-08-17 00:00:00   
1          2           1          1  2008/2009      1  2008-08-16 00:00:00   
2          3           1          1  2008/2009      1  2008-08-16 00:00:00   
3          4           1          1  2008/2009      1  2008-08-17 00:00:00   
4          5           1          1  2008/2009      1  2008-08-16 00:00:00   
...      ...         ...        ...        ...    ...                  ...   
25974  25975       24558      24558  2015/2016      9  2015-09-22 00:00:00   
25975  25976       24558      24558  2015/2016      9  2015-09-23 00:00:00   
25976  25977       24558      24558  2015/2016      9  2015-09-23 00:00:00   
25977  25978       24558      24558  2015/2016      9  2015-09-22 00:00:00   
25978  25979       24558      24558  2015/2016      9  2015-09-23 00:00:00   

       match_api_id  home_team_api_id  away_team_api_id  home_team_goal  ...  \
0            492473              9987              9993               1  ...   
1            492474             10000              9994               0  ...   
2            492475              9984              8635               0  ...   
3            492476              9991              9998               5  ...   
4            492477              7947              9985               1  ...   
...             ...               ...               ...             ...  ...   
25974       1992091             10190             10191               1  ...   
25975       1992092              9824             10199               1  ...   
25976       1992093              9956             10179               2  ...   
25977       1992094              7896             10243               0  ...   
25978       1992095             10192              9931               4  ...   

        SJA   VCH   VCD   VCA   GBH   GBD   GBA   BSH   BSD   BSA  
0      4.00  1.65  3.40  4.50  1.78  3.25  4.00  1.73  3.40  4.20  
1      3.80  2.00  3.25  3.25  1.85  3.25  3.75  1.91  3.25  3.60  
2      2.50  2.35  3.25  2.65  2.50  3.20  2.50  2.30  3.20  2.75  
3      7.50  1.45  3.75  6.50  1.50  3.75  5.50  1.44  3.75  6.50  
4      1.73  4.50  3.40  1.65  4.50  3.50  1.65  4.75  3.30  1.67  
...     ...   ...   ...   ...   ...   ...   ...   ...   ...   ...  
25974   NaN   NaN   NaN   NaN   NaN   NaN   NaN   NaN   NaN   NaN  
25975   NaN   NaN   NaN   NaN   NaN   NaN   NaN   NaN   NaN   NaN  
25976   NaN   NaN   NaN   NaN   NaN   NaN   NaN   NaN   NaN   NaN  
25977   NaN   NaN   NaN   NaN   NaN   NaN   NaN   NaN   NaN   NaN  
25978   NaN   NaN   NaN   NaN   NaN   NaN   NaN   NaN   NaN   NaN  

[25979 rows x 115 columns]

**2.4 League Table**

In [34]:
GetTable_DataFrame("League")

id  country_id                      name
0       1           1    Belgium Jupiler League
1    1729        1729    England Premier League
2    4769        4769            France Ligue 1
3    7809        7809     Germany 1. Bundesliga
4   10257       10257             Italy Serie A
5   13274       13274    Netherlands Eredivisie
6   15722       15722        Poland Ekstraklasa
7   17642       17642  Portugal Liga ZON Sagres
8   19694       19694   Scotland Premier League
9   21518       21518           Spain LIGA BBVA
10  24558       24558  Switzerland Super League

**2.5 Country Table**

In [35]:
GetTable_DataFrame("Country")

id         name
0       1      Belgium
1    1729      England
2    4769       France
3    7809      Germany
4   10257        Italy
5   13274  Netherlands
6   15722       Poland
7   17642     Portugal
8   19694     Scotland
9   21518        Spain
10  24558  Switzerland

**2.6 Team Table**

In [36]:
GetTable_DataFrame("Team")

id  team_api_id  team_fifa_api_id      team_long_name team_short_name
0        1         9987             673.0            KRC Genk             GEN
1        2         9993             675.0        Beerschot AC             BAC
2        3        10000           15005.0    SV Zulte-Waregem             ZUL
3        4         9994            2007.0    Sporting Lokeren             LOK
4        5         9984            1750.0   KSV Cercle Brugge             CEB
..     ...          ...               ...                 ...             ...
294  49479        10190             898.0       FC St. Gallen             GAL
295  49837        10191            1715.0             FC Thun             THU
296  50201         9777             324.0         Servette FC             SER
297  50204         7730            1862.0  FC Lausanne-Sports             LAU
298  51606         7896               NaN              Lugano             LUG

[299 rows x 5 columns]

**2.7 Team Attributes Table**

In [37]:
GetTable_DataFrame("Team_Attributes")

id  team_fifa_api_id  team_api_id                 date  \
0        1               434         9930  2010-02-22 00:00:00   
1        2               434         9930  2014-09-19 00:00:00   
2        3               434         9930  2015-09-10 00:00:00   
3        4                77         8485  2010-02-22 00:00:00   
4        5                77         8485  2011-02-22 00:00:00   
...    ...               ...          ...                  ...   
1453  1454             15005        10000  2011-02-22 00:00:00   
1454  1455             15005        10000  2012-02-22 00:00:00   
1455  1456             15005        10000  2013-09-20 00:00:00   
1456  1457             15005        10000  2014-09-19 00:00:00   
1457  1458             15005        10000  2015-09-10 00:00:00   

      buildUpPlaySpeed buildUpPlaySpeedClass  buildUpPlayDribbling  \
0                   60              Balanced                   NaN   
1                   52              Balanced                  48.0   
2                   47              Balanced                  41.0   
3                   70                  Fast                   NaN   
4                   47              Balanced                   NaN   
...                ...                   ...                   ...   
1453                52              Balanced                   NaN   
1454                54              Balanced                   NaN   
1455                54              Balanced                   NaN   
1456                54              Balanced                  42.0   
1457                54              Balanced                  42.0   

     buildUpPlayDribblingClass  buildUpPlayPassing buildUpPlayPassingClass  \
0                       Little                  50                   Mixed   
1                       Normal                  56                   Mixed   
2                       Normal                  54                   Mixed   
3                       Little                  70                    Long   
4                       Little                  52                   Mixed   
...                        ...                 ...                     ...   
1453                    Little                  52                   Mixed   
1454                    Little                  51                   Mixed   
1455                    Little                  51                   Mixed   
1456                    Normal                  51                   Mixed   
1457                    Normal                  51                   Mixed   

      ... chanceCreationShooting  chanceCreationShootingClass  \
0     ...                     55                       Normal   
1     ...                     64                       Normal   
2     ...                     64                       Normal   
3     ...                     70                         Lots   
4     ...                     52                       Normal   
...   ...                    ...                          ...   
1453  ...                     53                       Normal   
1454  ...                     50                       Normal   
1455  ...                     32                       Little   
1456  ...                     32                       Little   
1457  ...                     32                       Little   

     chanceCreationPositioningClass  defencePressure defencePressureClass  \
0                         Organised               50               Medium   
1                         Organised               47               Medium   
2                         Organised               47               Medium   
3                         Organised               60               Medium   
4                         Organised               47               Medium   
...                             ...              ...                  ...   
1453                      Organised               46               Medium   
1454                      Organised     

**2.8 Analysizing**
1. All tables have primary key that is called `id`